### PR TITLE
[SDK][Internal] Call `start_trace()` for flow test when internal enabled

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
- Fix line run evaluations calculation.
- Call start_trace for flow test.

Example for experiment test:
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py [hunk 1]
```diff
@@ -136,6 +139,9 @@ def _assert_result(result):
             assert expected_output_path.resolve().exists()
             # Assert eval metric exists
             assert (expected_output_path / "eval" / "flow.metrics.json").exists()
+            # Assert session exists
+            line_runs = client._traces.list_line_runs(session_id=session)
+            assert len(line_runs) > 0
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py:144]: test session also in non-experiment test?

Comment (brynn-code) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py]: Yes, and only for user who set the 'enable_internal_features'=True for now.

File: src/promptflow/promptflow/_trace/_start_trace.py [hunk 2]
```diff
@@ -62,7 +67,7 @@ def start_trace(*, session: typing.Optional[str] = None, **kwargs):
     inject_openai_api()
     # print user the UI url
     ui_url = f"http://localhost:{pfs_port}/v1.0/ui/traces?session={session_id}"
-    print(f"You can view the trace from UI url: {ui_url}")
+    _logger.info(f"You can view the trace from UI url: {ui_url}")
```
Comment (brynn-code) [inline src/promptflow/promptflow/_trace/_start_trace.py:70]: Double confirm: I assume we shall not use 'print' here

Comment (wangchao1230) [reply src/promptflow/promptflow/_trace/_start_trace.py:70]: As long as we can view it in notebook

**Review by wangchao1230 — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**